helix-vcs/src/git.rs

```diff
@@ -76,29 +78,20 @@ impl DiffProvider for Git {
         let file_oid = find_file_in_commit(&repo, &head, file)?;
 
         let file_object = repo.find_object(file_oid)?;
-        let mut data = file_object.detach().data;
-        // convert LF to CRLF if configured to avoid showing every line as changed
-        if repo
-            .config_snapshot()
-            .boolean("core.autocrlf")
-            .unwrap_or(false)
-        {
-            let mut normalized_file = Vec::with_capacity(data.len());
-            let mut at_cr = false;
-            for &byte in &data {
-                if byte == b'\n' {
-                    // if this is a LF instead of a CRLF (last byte was not a CR)
-                    // insert a new CR to generate a CRLF
-                    if !at_cr {
-                        normalized_file.push(b'\r');
-                    }
-                }
-                at_cr = byte == b'\r';
-                normalized_file.push(byte)
-            }
-            data = normalized_file
+        let data = file_object.detach().data;
+        // Get the actual data that git would make out of the git object.
+        // This will apply the user's git config or attributes like crlf conversions.
+        if let Some(work_dir) = repo.work_dir() {
+            let rela_path = file.strip_prefix(work_dir)?.to_string_lossy();
+            let (mut pipeline, _) = repo.filter_pipeline(None)?;
+            let mut worktree_outcome =
+                pipeline.convert_to_worktree(&data, rela_path.as_ref().into(), Delay::Forbid)?;
```

Comment:
@Byron how does this handle encoding? Git stores everything as utf-8 internally IIRC and only converts back on checkout.

I guess that this function would do that? Could we instead checkout utf-8 somehow. (My memory is very cloudy here so I may be missremebering)

Reply:
That's not the case, but it's true that Git has an 'internal' form that it wants to store in the object database. This is merely a form that passed through the filter pipeline though, which on Unix is most often a no-op.

It's true though that this method takes care of transforming `data` which is supposed to be directly from the object database and turn it into what would be checked out, applying all necessary transformations.